source/dagger/gameplay/platformer/character_controller_fsm.cpp

```diff
@@ -0,0 +1,61 @@
+#include "character_controller_fsm.h"
+#include "core/core.h"
+#include "core/engine.h"
+#include "core/input/inputs.h"
+#include "core/graphics/sprite.h"
+#include "core/graphics/animation.h"
+#include "gameplay/platformer/platformer_controller.h"
+
+using namespace dagger;
+
+// Idle
+
+
+void CharacterControllerFSM::Idle::Enter(CharacterControllerFSM::StateComponent& state_)
+{
+	auto& animator = Engine::Registry().get<Animator>(state_.entity);
+	AnimatorPlay(animator, "souls_like_knight_character:IDLE");
+}
+
+DEFAULT_EXIT(CharacterControllerFSM, Idle);
```

Comment:
maybe it'll be better to add flags in DEFINE_STATE, to add default implementation.
not sure if this will be better thou, so it's just an idea

Reply:
this way, if you want to add a non-default implementation, you change one file, not two, but othwerise is the same